### PR TITLE
filter deployment by applicaiton name

### PR DESCRIPTION
**Pull request by SiweiWang**

### CODE REVIEW

File: routes/deployments.js [hunk 1]
```diff
@@ -8,10 +8,31 @@ var Deployment = dbschema.Deployment;
 
 /* Gets all deployments. */
 router.get('/', function(req, res, next) {
-  Deployment.find(function (err, deployment) {
-    if (err) return next(err);
-    res.json(deployment);
-  });
+  var filter= {};
+  if (req && req.query && req.query.application){
+    var appliction = Application.findOne( {'name': req.query.application });
```
Comment (c4po) [inline routes/deployments.js]: can we run query without call back?

from http://mongoosejs.com/docs/queries.html
When a callback function:
- is passed, the operation will be executed immediately with the results passed to the callback.
- is not passed, an instance of Query is returned, which provides a special query builder interface.

**Comment by SiweiWang** [comment]
Yes, seems I need to run .exec to it . right now it is just a place holder for this feature . It should not break anything . I am thinking use aync.js to avoid this callback hell

**Comment by SiweiWang** [comment]
Update the PR as per our chat over the phone

**Comment by c4po** [comment]
looks good

**Pull request merged**